src/lib.rs

```diff
@@ -262,7 +262,10 @@ mod signature;
 mod signing;
 mod verifying;
 
+#[cfg(feature = "digest")]
 pub use curve25519_dalek::digest::Digest;
+#[cfg(feature = "digest")]
+pub use sha2::Sha512;
```

Comment:
If we are not supposed to re-export Sha512 then I need to remove Sha512 re-export here.
If we are supposed to re-export Sha512 then I need to adjust some integration tests to refer our re-export